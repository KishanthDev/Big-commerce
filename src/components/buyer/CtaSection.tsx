import { ArrowRight, Play, Check } from 'lucide-react';

export default function CtaSection() {
  const benefits = [
    'Free shipping on orders over $30',
    'Easy returns within 30 days',
    'Secure payment processing',
    'Direct communication with sellers',
    '24/7 customer support'
  ];

  return (
    <section className="bg-blue-600 dark:bg-black text-white py-16">
      <div className="container mx-auto px-5">
        <div className="flex flex-col lg:flex-row gap-10">
          <div className="lg:w-2/3">
            <h2 className="text-3xl font-semibold mb-4">Ready to Discover Amazing Products?</h2>
            <p className="text-lg opacity-90 mb-6 max-w-2xl">
              Join thousands of shoppers finding unique items from independent creators. Start shopping today with our 30-day satisfaction guarantee.
            </p>
            <div className="flex flex-col sm:flex-row gap-4">
              <button className="bg-white dark:bg-gray-300 text-blue-600 px-5 py-3 rounded-md font-medium hover:bg-gray-100 transition flex items-center justify-center">
                Start Shopping Today <ArrowRight className="w-4 h-4 ml-2" />
              </button>
              <button className="bg-transparent border border-white px-5 py-3 rounded-md font-medium dark:hover:bg-gray-600 hover:bg-white hover:bg-opacity-10 transition flex items-center justify-center">
                <Play className="w-4 h-4 mr-2" /> Watch Demo
              </button>
            </div>
          </div>
          <div className="lg:w-1/3 bg-white dark:bg-gray-300 rounded-lg p-6 text-gray-900">
            <h3 className="text-lg font-semibold mb-5">Shopper Benefits</h3>
            <ul className="space-y-3">
              {benefits.map((benefit, index) => (
                <li key={index} className="flex items-start">
                  <Check className="w-5 h-5 text-blue-600 mr-2 mt-0.5 flex-shrink-0" />
                  <span className="text-sm">{benefit}</span>
                </li>
              ))}
            </ul>
          </div>
        </div>
      </div>
    </section>
  );
}