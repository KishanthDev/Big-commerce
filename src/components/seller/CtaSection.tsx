import { ArrowRight, Play, Check } from "lucide-react";

export default function CtaSection() {
  const benefits = [
    "No commision fees",
    "Easy-to-use dashboard",
    "local customers targeting",
    "Secure payment processing",
    "24/7 seller support",
  ];

  return (
    <section
      role="region"
      data-testid="cta-section"
      className="bg-blue-600 dark:bg-black text-white py-16"
    >
      <div data-testid="cta-container" className="container mx-auto px-5">
        <div className="flex flex-col lg:flex-row gap-10">
          <div className="lg:w-2/3">
            <h2 className="text-3xl font-semibold mb-4">
              Ready to Discover Amazing Products?
            </h2>
            <p className="text-lg opacity-90 mb-6 max-w-2xl">
              Join thousands of sellers who are thriving on our platform.Start
              selling today with our risk-free 30-day trail
            </p>
            <div className="flex flex-col sm:flex-row gap-4">
              <button className="bg-white dark:bg-gray-300 text-blue-600 px-5 py-3 rounded-md font-medium hover:bg-gray-300 transition flex items-center justify-center">
                Start Selling Today <ArrowRight className="w-4 h-4 ml-2" />
              </button>
              <button className="bg-transparent border border-white px-5 py-3 rounded-md font-medium dark:hover:bg-gray-600 hover:bg-gray-300 hover:bg-opacity-10 transition flex items-center justify-center">
                <Play className="w-4 h-4 mr-2" /> Watch Demo
              </button>
            </div>
          </div>
          <div
            data-testid="benefits-card"
            className="lg:w-1/3 bg-white dark:bg-gray-300 rounded-lg p-6 text-gray-900"
          >
            <h3 className="text-lg font-semibold mb-5">Shopper Benefits</h3>
            <ul className="space-y-3">
              {benefits.map((benefit, index) => (
                <li key={index} className="flex items-start">
                  <Check className="w-5 h-5 text-blue-600 mr-2 mt-0.5 flex-shrink-0" />
                  <span className="text-sm">{benefit}</span>
                </li>
              ))}
            </ul>
          </div>
        </div>
      </div>
    </section>
  );
}
