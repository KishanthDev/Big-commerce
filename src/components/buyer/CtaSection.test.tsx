import { render, screen, fireEvent } from '@testing-library/react';
import CtaSection from './CtaSection';
import { ArrowRight, Play, Check } from 'lucide-react';

// Mock lucide-react icons
jest.mock('lucide-react', () => ({
  ArrowRight: jest.fn(() => <svg data-testid="arrow-right" />),
  Play: jest.fn(() => <svg data-testid="play" />),
  Check: jest.fn(() => <svg data-testid="check" />),
}));

describe('CtaSection', () => {
  const mockBenefits = [
    "Free shipping on orders over $30",
    "Easy returns within 30 days",
    "Secure payment processing",
    "Direct communication with sellers",
    "24/7 customer support",
  ];

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders without crashing', () => {
    render(<CtaSection />);
    expect(screen.getByText('Ready to Discover Amazing Products?')).toBeInTheDocument();
  });

  test('displays all benefits with check icons', () => {
    render(<CtaSection />);
    const benefitItems = screen.getAllByTestId('check');
    expect(benefitItems).toHaveLength(mockBenefits.length);
    mockBenefits.forEach(benefit => {
      expect(screen.getByText(benefit)).toBeInTheDocument();
    });
  });

  test('renders both action buttons correctly', () => {
    render(<CtaSection />);
    expect(screen.getByRole('button', { name: /Start Shopping Today/i })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /Watch Demo/i })).toBeInTheDocument();
  });

  test('includes correct icons in buttons', () => {
    render(<CtaSection />);
    expect(ArrowRight).toHaveBeenCalled();
    expect(Play).toHaveBeenCalled();
  });

  test('has proper accessibility attributes', () => {
    render(<CtaSection />);
    expect(screen.getByRole('heading', { level: 2 })).toHaveTextContent('Ready to Discover Amazing Products?');
    expect(screen.getByRole('heading', { level: 3 })).toHaveTextContent('Shopper Benefits');
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<CtaSection />);
    expect(asFragment()).toMatchSnapshot();
  });
});