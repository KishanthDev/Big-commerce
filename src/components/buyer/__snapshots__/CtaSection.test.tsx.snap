// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CtaSection matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="bg-blue-600 dark:bg-black text-white py-16"
  >
    <div
      class="container mx-auto px-5"
    >
      <div
        class="flex flex-col lg:flex-row gap-10"
      >
        <div
          class="lg:w-2/3"
        >
          <h2
            class="text-3xl font-semibold mb-4"
          >
            Ready to Discover Amazing Products?
          </h2>
          <p
            class="text-lg opacity-90 mb-6 max-w-2xl"
          >
            Join thousands of shoppers finding unique items from independent creators. Start shopping today with our 30-day satisfaction guarantee.
          </p>
          <div
            class="flex flex-col sm:flex-row gap-4"
          >
            <button
              class="bg-white dark:bg-gray-300 text-blue-600 px-5 py-3 rounded-md font-medium hover:bg-gray-300 transition flex items-center justify-center"
            >
              Start Shopping Today 
              <svg
                data-testid="arrow-right"
              />
            </button>
            <button
              class="bg-transparent border border-white px-5 py-3 rounded-md font-medium dark:hover:bg-gray-600 hover:bg-gary-300 hover:bg-opacity-10 transition flex items-center justify-center"
            >
              <svg
                data-testid="play"
              />
               Watch Demo
            </button>
          </div>
        </div>
        <div
          class="lg:w-1/3 bg-white dark:bg-gray-300 rounded-lg p-6 text-gray-900"
        >
          <h3
            class="text-lg font-semibold mb-5"
          >
            Shopper Benefits
          </h3>
          <ul
            class="space-y-3"
          >
            <li
              class="flex items-start"
            >
              <svg
                data-testid="check"
              />
              <span
                class="text-sm"
              >
                Free shipping on orders over $30
              </span>
            </li>
            <li
              class="flex items-start"
            >
              <svg
                data-testid="check"
              />
              <span
                class="text-sm"
              >
                Easy returns within 30 days
              </span>
            </li>
            <li
              class="flex items-start"
            >
              <svg
                data-testid="check"
              />
              <span
                class="text-sm"
              >
                Secure payment processing
              </span>
            </li>
            <li
              class="flex items-start"
            >
              <svg
                data-testid="check"
              />
              <span
                class="text-sm"
              >
                Direct communication with sellers
              </span>
            </li>
            <li
              class="flex items-start"
            >
              <svg
                data-testid="check"
              />
              <span
                class="text-sm"
              >
                24/7 customer support
              </span>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
